ddtrace/_trace/trace_handlers.py

```diff
@@ -144,6 +145,10 @@ def _start_span(ctx: core.ExecutionContext, call_trace: bool = True, **kwargs) -
         # dispatch event for inferred proxy finish
         core.dispatch("inferred_proxy.finish", (ctx,))
 
+    # Set to False so future calls, when referencing the parent context for activate_distributed_headers
+    # will not needlessly activate distributed headers.
+    ctx.set_item("activate_distributed_headers", False)
```

Comment:
This works. Another approach would be to use `core.get_local_item(...)` instead of `core.get_item(...)`. This way we only access the kwargs set on the current context. We could probably use this method for all the distributed tracing header kwargs

Reply:
Ah yeah, great find. I'll switch to using get_local_item.

Reply:
I'll make a follow up PR exploring what else we can switch to using local.